src/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -367,6 +372,26 @@ private void RemoveRestoreBar()
             }
         }
 
+        private void AddRestartRequestBar(IVsShell4 vsRestarter)
+        {
+            if (Model.Context.PackageManager.DeleteOnRestartManager != null)
+            {
+                _restartBar = new RestartRequestBar(Model.Context.PackageManager.DeleteOnRestartManager, vsRestarter);
+                _root.Children.Add(_restartBar);
+            }
+        }
+
+        private void RemoveRestartBar()
+        {
+            if (_restartBar != null)
+            {
+                _restartBar.CleanUp();
+
+                // TODO: clean this up during dispose also
```

Comment:
Please remove the TODOs. File a bug or do what the comment intends. Same comment for the rest of the review